src/modules/esl-event-listener/core/descriptors.ts

```diff
@@ -0,0 +1,73 @@
+import {isObject} from '../../esl-utils/misc/object/types';
+
+import type {ESLListenerDescriptorExt, ESLListenerDescriptorFn} from './types';
+
+/** Key to store listeners on the host */
+const DESCRIPTORS = (window.Symbol || String)('__esl_descriptors');
+
+/**
+ * @param host - host object
+ * @param createIfNotExists - create keys store on the host if not exists
+ * @returns object own descriptors keys or an empty array
+ */
+function getOwnDescriptors(host: object, createIfNotExists = false): string[] {
+  if (Object.hasOwnProperty.call(host, DESCRIPTORS)) return (host as any)[DESCRIPTORS];
+  const value: string[]  = [];
+  if (createIfNotExists) Object.defineProperty(host, DESCRIPTORS, {value, configurable: true});
```

Comment:
cosmetic: it would probably be a bit more transparent to define the property in the separate function

Reply:
Totally agree. Moved to release 4.3.0 enhancements